lgsm/functions/command_monitor.sh

```diff
@@ -92,48 +95,43 @@ fn_query_gsquery(){
 }
 
 fn_query_tcp(){
-	bash -c 'exec 3<> /dev/tcp/'${queryip}'/'${queryport}'' > /dev/null 2>&1
+	bash -c "exec 3<> '/dev/tcp/${queryip}/${queryport}'" > /dev/null 2>&1
 	querystatus="$?"
 }
 
 fn_monitor_query(){
-# Will loop and query up to 5 times every 15 seconds.
-# Query will wait up to 60 seconds to confirm server is down as server can become non-responsive during map changes.
-totalseconds=0
-for queryattempt in {1..5}; do
-	for queryip in "${queryips[@]}"; do
-		fn_print_dots "Querying port: ${querymethod}: ${queryip}:${queryport} : ${totalseconds}/${queryattempt}: "
-		fn_print_querying_eol
-		fn_script_log_info "Querying port: ${querymethod}: ${queryip}:${queryport} : ${queryattempt} : QUERYING"
-		# querydelay
-		if [ "$(head -n 1 "${lockdir}/${selfname}.lock")" -gt "$(date "+%s" -d "${querydelay} mins ago")" ]; then
-			fn_print_ok "Querying port: ${querymethod}: ${ip}:${queryport} : ${totalseconds}/${queryattempt}: "
-			fn_print_delay_eol_nl
-			fn_script_log_info "Querying port: ${querymethod}: ${ip}:${queryport} : ${queryattempt} : DELAY"
-			fn_script_log_info "Query bypassed: ${gameservername} started less than ${querydelay} minutes ago"
-			fn_script_log_info "Server started: $(date -d @$(head -n 1 "${lockdir}/${selfname}.lock"))"
-			fn_script_log_info "Current time: $(date)"
-			monitorpass=1
-			core_exit.sh
-		# will use query method selected in fn_monitor_loop
-		# gamedig
-		elif [ "${querymethod}" ==  "gamedig" ]; then
-			query_gamedig.sh
-		# gsquery
-		elif [ "${querymethod}" ==  "gsquery" ]; then
-			fn_query_gsquery
-		#tcp query
-		elif [ "${querymethod}" ==  "tcp" ]; then
-			fn_query_tcp
-		fi
+	local fail_after="60" # seconds
+	local time_per_attempt="3"
+	local max_attempts="5"
+	local wait_between_attempts="$(( (fail_after-max_attempts*time_per_attempt) / (max_attempts-1) ))"
+
+	# Will loop and query up to 5 times every 15 seconds.
+	# Query will wait up to 60 seconds to confirm server is down as server can become non-responsive during map changes.
+	for queryattempt in $(seq 1 "${max_attempts}" ); do
 
-		if [ "${querystatus}" == "0" ]; then
-			# Server query OK.
-			fn_print_ok "Querying port: ${querymethod}: ${queryip}:${queryport} : ${totalseconds}/${queryattempt}: "
-			fn_print_ok_eol_nl
-			fn_script_log_pass "Querying port: ${querymethod}: ${queryip}:${queryport} : ${queryattempt}: OK"
-			monitorpass=1
+		for queryip in "${queryips[@]}"; do
+			local log_msg="Starting to query in mode \"${querymethod}\" to target \"${queryip}:${queryport}\" attempt ${queryattempt} / ${max_attempts}"
+			fn_print_dots "${log_msg}"
+
+			# will use query method selected in fn_monitor_loop
+			querystatus="100"
+			if [ "${querymethod}" ==  "gamedig" ]; then
+				query_gamedig.sh
+
```

Comment:
cr not required here

Reply:
removed. Same readability intention as above.